### PR TITLE
Miscellaneous improvements done during finite cycle group implementation

### PR DESCRIPTION
**Pull request by uzytkownik**
One of the sub-pull requests split from PR #1076.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -254,6 +254,12 @@ attached to all deprecated names to discourage their use.
   Any¬→¬All  ↦  Any¬⇒¬All
   ```
 
+* In `Data.Nat.Properties
+  ```agda
+  ∀[m≤n⇒m≢o]⇒o<n  ↦  ∀[m≤n⇒m≢o]⇒n<o
+  ∀[m<n⇒m≢o]⇒o≤n  ↦  ∀[m<n⇒m≢o]⇒n≤o
+  ```
```
Comment (MatthewDaggitt) [inline CHANGELOG.md]: Nice spot, thanks! I think this should be listed under the "bugs" section at the top.

File: src/Data/Nat/DivMod.agda [hunk 2]
```diff
@@ -115,6 +115,26 @@ m<[1+n%d]⇒m≤[n%d] {m} n (suc d-1) = k<1+a[modₕ]n⇒k≤a[modₕ]n 0 m n d-
   (m % d +  n % d + (n / d) * d)  % d ≡⟨ [m+kn]%n≡m%n (m % d + n % d) (n / d) d-1 ⟩
   (m % d +  n % d)                % d ∎
 
+%-distribˡ-* : ∀ m n d {≢0} → ((m * n) % d) {≢0} ≡ (((m % d) {≢0} * (n % d) {≢0}) % d) {≢0}
+%-distribˡ-* m n d@(suc d-1) = begin-equality
+  (m * n)                                             % d ≡⟨ cong (λ h → (h * n) % d) (m≡m%n+[m/n]*n m d-1) ⟩
+  ((m′ + k * d) * n)                                  % d ≡⟨ cong (λ h → h % d) (*-distribʳ-+ n m′ (k * d)) ⟩
+  (m′ * n + (k * d) * n)                              % d ≡⟨ cong (λ h → (m′ * h + (k * d) * h) % d) (m≡m%n+[m/n]*n n d-1) ⟩
+  (m′ * (n′ + j * d) + (k * d) * (n′ + j * d))        % d ≡⟨ cong (λ h → (h + (k * d) * (n′ + j * d)) % d) (*-distribˡ-+ m′ n′ (j * d)) ⟩
```
Comment (MatthewDaggitt) [inline src/Data/Nat/DivMod.agda]: From here onwards should be solvable automatically using the new `Data.Nat.Tactic.RingSolver` module. Although you have may have to expand out the variables in the `where` block. See `README.Tactic.RingSolver` for details on how to use.

File: src/Data/Nat/DivMod.agda [hunk 3]
```diff
@@ -115,6 +115,26 @@ m<[1+n%d]⇒m≤[n%d] {m} n (suc d-1) = k<1+a[modₕ]n⇒k≤a[modₕ]n 0 m n d-
...
+  ((m′ * n′ + m′ * (j * d)) + (k * d) * (n′ + j * d)) % d ≡⟨ cong (λ h → h % d) (+-assoc (m′ * n′) (m′ * (j * d)) ((k * d) * (n′ + j * d))) ⟩
+  (m′ * n′ + (m′ * (j * d) + (k * d) * (n′ + j * d))) % d ≡˘⟨ cong (λ h → (m′ * n′ + (h + (k * d) * (n′ + j * d))) % d) (*-assoc m′ j d) ⟩
+  (m′ * n′ + ((m′ * j) * d + (k * d) * (n′ + j * d))) % d ≡⟨ cong (λ h → (m′ * n′ + ((m′ * j) * d + h)) % d) (*-comm (k * d) (n′ + j * d)) ⟩
+  (m′ * n′ + ((m′ * j) * d + (n′ + j * d) * (k * d))) % d ≡˘⟨ cong (λ h → (m′ * n′ + ((m′ * j) * d + h)) % d) (*-assoc (n′ + j * d) k d) ⟩
+  (m′ * n′ + ((m′ * j) * d + ((n′ + j * d) * k) * d)) % d ≡˘⟨ cong (λ h → (m′ * n′ + h) % d) (*-distribʳ-+ d (m′ * j) ((n′ + j * d) * k)) ⟩
+  (m′ * n′ + (m′ * j + (n′ + j * d) * k) * d)         % d ≡⟨ [m+kn]%n≡m%n (m′ * n′) (m′ * j + (n′ + j * d) * k) d-1 ⟩
+  (m′ * n′)                                           % d ≡⟨⟩
+  ((m % d) * (n % d)) % d ∎
+  where
+    m′ = m % d
```
Comment (MatthewDaggitt) [inline src/Data/Nat/DivMod.agda]: We don't usually put any indentation after `where` blocks.

File: src/Data/Nat/Properties.agda [hunk 4]
```diff
@@ -225,6 +225,9 @@ n≤1+n _ = ≤-step ≤-refl
 n≤0⇒n≡0 : ∀ {n} → n ≤ 0 → n ≡ 0
 n≤0⇒n≡0 z≤n = refl
 
+1+n≤1+m⇒n≤m : ∀ {m n} → suc m ≤ suc n → m ≤ n
+1+n≤1+m⇒n≤m (s≤s le) = le
```
Comment (MatthewDaggitt) [inline src/Data/Nat/Properties.agda]: As @gallais  mentioned this already exists as `≤-pred`

File: CHANGELOG.md (hunk 1 above)
Comment (uzytkownik) [reply CHANGELOG.md]: Done.

File: src/Data/Nat/DivMod.agda (hunk 2 above)
Comment (uzytkownik) [reply src/Data/Nat/DivMod.agda]: I added lemma. I could not get `solve` to work - only `solve-∀`.
I'd note that it doesn't show up in https://agda.github.io/agda-stdlib/README.html.

Comment (MatthewDaggitt) [reply src/Data/Nat/DivMod.agda]: Thanks, filed as #1083.

**Comment by MatthewDaggitt** [comment]
Thanks for the PR! Merged in.

**Pull request merged**